### PR TITLE
Validator v3

### PR DESCRIPTION
**Pull request by ratneshraval**
#### Which User story does this relate to?
- https://www.pivotaltracker.com/story/show/151420115
- https://www.pivotaltracker.com/story/show/151302409

#### Brief feature description
- validator update. Errors added to state
- phone number field allow to backspace when cursor is after area code.

### CODE REVIEW

**Comment by anuroop15** [comment]
screen shot 2017-09-26 at 9 28 00 am

@ratneshraval : was this "Unknown" wrapper tag over input created by Cleave.js ?? Can this be named according to type of field it's wrapping.?

**Comment by anuroop15** [comment]
screen shot 2017-09-26 at 9 31 56 am

good work on the bug to delete numbers before parentheses ().  I have identified that  hitting backspace once removes 2 numbers at a time when removing (123)- ___-___ 3 from the field.

**Comment by ratneshraval** [comment]
@anuroop15 
1. unknown tag is only under react code tab. If you look actual HTML, there is no unknown tag. 
2. I am aware of this bug and that is a problem with cleave (https://github.com/nosir/cleave.js/issues/234). I think this bug is better than not allowing to delete at all. We will have a bug story for that where we can research more and fix.

File: app/javascript/components/common/phoneNumberFields.js [hunk 1]
```diff
@@ -5,44 +5,44 @@ import {DropDownField} from './dropDownField'
...
 
-    const phoneNumberId = this.props.idPrefix + 'number'
-
     return (
       <div>
 
         <CleaveInputField
           gridClassName='col-md-4'
-          id={phoneNumberId}
+          id={this.phoneNumberId}
```
Comment (lauherkcambria) [inline app/javascript/components/common/phoneNumberFields.js]: why make this change?

Comment (ratneshraval) [reply app/javascript/components/common/phoneNumberFields.js]: so I don't have to build phoneNumberId twice. Before we were building the string in constructor and render. Now I build it in constructor and use it in render.

**Pull request merged**